### PR TITLE
play2.6

### PR DESCRIPTION
**Pull request by Tateni93**

### CODE REVIEW

File: app/controllers/EligibilityController.scala [hunk 1]
```diff
@@ -27,21 +27,21 @@ import forms.MultiYearLowerEarnerForm.lowerEarnerForm
 import forms.MultiYearPartnersIncomeQuestionForm.partnersIncomeForm
 import javax.inject.Inject
 import models.Country
-import play.api.i18n.MessagesApi
-import play.api.mvc.{Action, AnyContent, Call}
+import play.api.mvc.{Action, AnyContent, Call, MessagesControllerComponents}
```
Comment (Tateni93) [inline app/controllers/EligibilityController.scala:30]: Are we using all; these imports?

File: app/controllers/TransferController.scala [hunk 2]
```diff
@@ -111,7 +108,7 @@ class TransferController @Inject()(
         case CurrentAndPreviousYearsEligibility(false, Nil, _, _) =>
           throw new NoTaxYearsAvailable
         case CurrentAndPreviousYearsEligibility(false, previousYears, registrationInput, _)
-          if previousYears.nonEmpty => Ok(views.html.multiyear.transfer.previous_years(registrationInput, previousYears, false))
+          if previousYears.nonEmpty => Ok(views.html.multiyear.transfer.previous_years(registrationInput, previousYears, currentYearAvailable = false))
```
Comment (Tateni93) [inline app/controllers/TransferController.scala:111]: Don't need to name currentYearAvailable here, there are only 3 params in this view.

File: app/controllers/EligibilityController.scala (hunk 1 above)
Comment (Tateni93) [reply app/controllers/EligibilityController.scala:30]: Yeah

File: app/controllers/TransferController.scala (hunk 2 above)
Comment (Tateni93) [reply app/controllers/TransferController.scala:111]: Ignore me, good practice to name boolean parameters

File: app/controllers/actions/UnauthenticatedActionTransformer.scala [hunk 3]
```diff
@@ -28,10 +28,14 @@ import uk.gov.hmrc.play.HeaderCarrierConverter
...
 class UnauthenticatedActionTransformer @Inject()(
-                                                  val authConnector: AuthConnector
+                                                  val authConnector: AuthConnector,
+                                                    cc: MessagesControllerComponents
                                                 )(implicit ec: ExecutionContext)
   extends ActionTransformer[Request, UserRequest]
-    with ActionBuilder[UserRequest] with AuthorisedFunctions {
+    with ActionBuilder[UserRequest, AnyContent] with AuthorisedFunctions {
+
+  override val parser: BodyParser[AnyContent] = cc.parsers.defaultBodyParser
```
[the remark this answers is not in the document]
Comment (Tateni93) [reply app/controllers/actions/UnauthenticatedActionTransformer.scala:37]: Done

File: app/controllers/actions/UnauthenticatedActionTransformer.scala [hunk 4]
```diff
@@ -28,10 +28,14 @@ import uk.gov.hmrc.play.HeaderCarrierConverter
...
+  override protected val executionContext: ExecutionContext = cc.executionContext
```
[the remark this answers is not in the document]
Comment (Tateni93) [reply app/controllers/actions/UnauthenticatedActionTransformer.scala:38]: Done

File: app/controllers/actions/UnauthenticatedActionTransformer.scala [hunk 5]
```diff
@@ -28,10 +28,14 @@ import uk.gov.hmrc.play.HeaderCarrierConverter
 import scala.concurrent.{ExecutionContext, Future}
 
 class UnauthenticatedActionTransformer @Inject()(
-                                                  val authConnector: AuthConnector
+                                                  val authConnector: AuthConnector,
+                                                    cc: MessagesControllerComponents
                                                 )(implicit ec: ExecutionContext)
```
[the remark this answers is not in the document]
Comment (Tateni93) [reply app/controllers/actions/UnauthenticatedActionTransformer.scala:33]: Done

File: app/errorHandler/ErrorHandler.scala [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+import play.api.Configuration
+import play.api.i18n.MessagesApi
+import play.api.mvc.Request
+import play.twirl.api.Html
+import uk.gov.hmrc.play.bootstrap.frontend.http.FrontendErrorHandler
+import uk.gov.hmrc.renderer.TemplateRenderer
+
+class ErrorHandler @Inject()(
+                             val messagesApi: MessagesApi,
+                             val configuration: Configuration,
```
[the remark this answers is not in the document]
Comment (Tateni93) [reply app/errorHandler/ErrorHandler.scala:30]: Done

File: app/metrics/Metrics.scala [hunk 7]
```diff
@@ -17,26 +17,18 @@
...
-trait Metrics {
-  def incrementSuccessCitizenDetail(): Unit
-
-  def incrementFailedCitizenDetail(): Unit
-
-  def citizenDetailStartTimer(): Timer.Context
-}
-
-object Metrics extends Metrics with MicroserviceMetrics {
+@Singleton
```
[the remark this answers is not in the document]
Comment (Tateni93) [reply app/metrics/Metrics.scala:23]: Removed

File: test/models/RelationshipRecordTest.scala [hunk 8]
```diff
@@ -94,13 +94,13 @@ class RelationshipRecordTest extends UnitSpec {
       "participant endDate is in PastYear" in {
         val relationshipRecord = relationshipActiveRecordWithPastValidDate
 
-        relationshipRecord.overlappingTaxYears shouldBe Set(2012, 2013, 2014, 2015, 2016, 2017, 2018)
+        relationshipRecord.overlappingTaxYears shouldBe Set(2012, 2013, 2014, 2015, 2016, 2017, 2019, 2018)
```
[the remark this answers is not in the document]
Comment (Tateni93) [reply test/models/RelationshipRecordTest.scala:97]: We can take a look at this as part of the test refactor element of this process

File: app/models/CurrentAndPreviousYearsEligibility.scala [hunk 9]
```diff
@@ -16,21 +16,15 @@
...
-}
-
 case class CurrentAndPreviousYearsEligibility(currentYearAvailable: Boolean, previousYears: List[TaxYear],
                                               registrationInput: RegistrationFormInput, availableTaxYears: List[TaxYear])
 
-object CurrentAndPreviousYearsEligibility extends YearsEligibility {
+object CurrentAndPreviousYearsEligibility {
 
-  val timeService = TimeService
+  val taxYear: Int = uk.gov.hmrc.time.TaxYear.current.startYear
```
[the remark this answers is not in the document]
Comment (Tateni93) [reply app/models/CurrentAndPreviousYearsEligibility.scala:24]: Done

File: app/services/TimeService.scala [hunk 10]
```diff
@@ -17,10 +17,12 @@
...
 
 import config.ApplicationConfig
-import org.joda.time.LocalDate
-import org.joda.time.format.DateTimeFormat
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
 import uk.gov.hmrc.time.TaxYear
 
+//TODO look into making this a util or deleting. It Is Not a Service!
```
[the remark this answers is not in the document]
Comment (Tateni93) [reply app/services/TimeService.scala:25]: This TODO can be addressed in the code refactor TIcket to be provided later

File: app/views/helpers/LanguageUtils.scala [hunk 11]
```diff
@@ -17,16 +33,20 @@
...
-import org.joda.time.LocalDate
-import org.joda.time.format.DateTimeFormat
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
 import play.api.data.Form
 import play.api.i18n.Messages
 import uk.gov.hmrc.time.TaxYear
 
+//TODO we may be able to delete this
```
[the remark this answers is not in the document]
Comment (Tateni93) [reply app/views/helpers/LanguageUtils.scala:45]: This ticket will be raised shortly for the refactor stage

File: conf/application.conf [hunk 12]
```diff
@@ -37,12 +53,12 @@ metrics {
   rateUnit = SECONDS
   durationUnit = SECONDS
   showSamples = true
-  jvm = true
+  jvm = false
```
[the remark this answers is not in the document]
Comment (Tateni93) [reply conf/application.conf:56]: Done

File: conf/application.conf [hunk 13]
```diff
@@ -37,12 +53,12 @@ metrics {
...
   enabled = true
 }
 
 auditing {
-  enabled = true
+  enabled = false
```
[the remark this answers is not in the document]
Comment (ryanniven) [reply conf/application.conf:61]: done

**Pull request merged**